src/sinks/blackhole/sink.rs

```diff
@@ -73,12 +71,7 @@ impl StreamSink<Event> for BlackholeSink {
                 raw_bytes_collected = total_raw_bytes.load(Ordering::Relaxed)
             }, "Total events collected");
         });
-        let chunk_size = match self.config.rate {
-            None => MAX_CHUNK_SIZE,
-            Some(rate) => rate.min(MAX_CHUNK_SIZE),
-        };
-        let mut chunks = input.ready_chunks(chunk_size);
-        while let Some(events) = chunks.next().await {
+        while let Some(events) = input.next().await {
```

Comment:
If I'm reading this right, this changes the rate limiting semantics slightly. Instead of limiting to exactly the number of events, it may pull more than the limit if a batch overlaps the limit, and then pause until the next batch?

We may want to update the docs to make that clear if so.